packages/gatsby/src/utils/webpack-utils.ts

```diff
@@ -43,84 +31,100 @@ type Stage = "develop" | "develop-html" | "build-javascript" | "build-html"
-  yaml: LoaderResolver<*>,
-  null: LoaderResolver<*>,
-  raw: LoaderResolver<*>,
-
-  style: LoaderResolver<*>,
-  css: LoaderResolver<*>,
+export interface ILoaderUtils {
+  json: LoaderResolver
+  yaml: LoaderResolver
+  null: LoaderResolver
+  raw: LoaderResolver
+
+  style: LoaderResolver
+  css: LoaderResolver
   postcss: LoaderResolver<{
-    browsers?: string[],
-    plugins?: Array<any> | ((loader: any) => Array<any>),
-  }>,
+    browsers?: string[]
+    overrideBrowserslist?: string[]
+    plugins?: Array<any> | ((loader: any) => Array<any>)
+  }>
+
+  file: LoaderResolver
+  url: LoaderResolver
+  js: LoaderResolver
+  dependencies: LoaderResolver
 
-  file: LoaderResolver<*>,
-  url: LoaderResolver<*>,
-  js: LoaderResolver<*>,
-  dependencies: LoaderResovler<*>,
+  miniCssExtract: LoaderResolver
+  imports: LoaderResolver
+  exports: LoaderResolver
 
-  miniCssExtract: LoaderResolver<*>,
-  imports: LoaderResolver<*>,
-  exports: LoaderResolver<*>,
+  eslint: LoaderResolver
+}
 
-  eslint: LoaderResolver<*>,
+interface IModuleThatUseGatsby {
+  name: string
+  path: string
 }
 
 /**
  * Utils that produce webpack rule objects
  */
-export type RuleUtils = {
+export interface IRuleUtils {
   /**
    * Handles JavaScript compilation via babel
    */
-  js: RuleFactory<*>,
-  yaml: RuleFactory<*>,
-  fonts: RuleFactory<*>,
-  images: RuleFactory<*>,
-  miscAssets: RuleFactory<*>,
-
-  css: ContextualRuleFactory,
-  cssModules: RuleFactory<*>,
-  postcss: ContextualRuleFactory,
-
-  eslint: RuleFactory<*>,
+  js: RuleFactory<{ modulesThatUseGatsby?: IModuleThatUseGatsby[] }>
+  dependencies: RuleFactory
+  yaml: RuleFactory
+  fonts: RuleFactory
+  images: RuleFactory
+  miscAssets: RuleFactory
+  media: RuleFactory
+
+  css: ContextualRuleFactory<{ browsers?: string[] }>
+  cssModules: RuleFactory
+  postcss: ContextualRuleFactory<{ overrideBrowserOptions: any }>
+
+  eslint: RuleFactory
 }
 
-export type PluginUtils = BuiltinPlugins & {
-  extractText: PluginFactory,
-  uglify: PluginFactory,
-  moment: PluginFactory,
-  extractStats: PluginFactory,
-}
+export type PluginUtils = BuiltinPlugins &
+  Partial<{
+    extractText: PluginFactory
+    uglify: PluginFactory
+    moment: PluginFactory
+    extractStats: PluginFactory
+    minifyJs: PluginFactory
+    minifyCss: PluginFactory
+    fastRefresh: PluginFactory
+    eslintGraphqlSchemaReload: PluginFactory
+  }>
 
 /**
  * webpack atoms namespace
  */
-export type WebpackUtils = {
-  loaders: LoaderUtils,
+export interface IWebpackUtils {
+  loaders: ILoaderUtils
 
-  rules: RuleUtils,
+  rules: Partial<IRuleUtils>
 
-  plugins: PluginUtils,
+  plugins: PluginUtils
 }
 
 /**
  * A factory method that produces an atoms namespace
  */
-module.exports = async ({
+export const createUtils = async ({
   stage,
   program,
 }: {
-  stage: Stage,
-  program: any,
-}): Promise<WebpackUtilsOptions> => {
+  stage: Stage
+  program: any
+}): Promise<IWebpackUtils> => {
```

Comment:
so I looked into this, there is no asynchroneous code in this function. We can drop the async and the Promise type